tests/test_processor_composition.py

```diff
@@ -115,6 +115,27 @@ def test_remote_processor_creation(requests_mock):
     rp.add(0, BS())
 
 
+def test_remote_processor_composition_input_ports(requests_mock):
+    ls = LogicalState([0, 0])
+    cnot = catalog['postprocessed cnot'].build_processor()
+    cnot.min_detected_photons_filter(2)
+
+    rp = RemoteProcessor(rpc_handler=get_rpc_handler(requests_mock), m=4)
+    rp.min_detected_photons_filter(2)
+    rp.add(0, cnot)
+    rp.with_input(ls)
+
+    # check that the input ports of the cnot are identical to the remote processor
+    for mode in range(4):
+        cnot_input_port = cnot.get_input_port(mode)
+        rp_input_port = rp.get_input_port(mode)
+        assert cnot_input_port == rp_input_port
+
+        cnot_output_port = cnot.get_output_port(mode)
```

Comment:
The ticket only asks for the input ports, so here you are testing something that you didn't change (my bad).
Since there were no tests for the output ports, you can keep it though.